libhwmon/wrapadl.cpp

```diff
@@ -192,12 +255,12 @@ int wrap_adl_get_tempC(wrap_adl_handle *adlh, int gpuindex, unsigned int *tempC)
 int wrap_adl_get_fanpcnt(wrap_adl_handle *adlh, int gpuindex, unsigned int *fanpcnt)
 {
 	wrap_adlReturn_t rc;
-	if (gpuindex < 0 || gpuindex >= adlh->adl_gpucount)
+	if (gpuindex < 0 || gpuindex >= adlh->log_gpucount){
 		return -1;
-
+	}
 	ADLFanSpeedValue *fan = new ADLFanSpeedValue();
 	fan->iSpeedType = 1;
-	rc = adlh->adlOverdrive5FanSpeedGet(adlh->phys_logi_device_id[gpuindex], 0, fan);
+	rc = adlh->adlOverdrive5FanSpeedGet(gpuindex, 0, fan);
```

Comment:
phys_logi_device_id  maps to the first logical adapter so this needs to be used too